python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -207,12 +228,99 @@ def parse_stream_tables(
+            details.
+        workspace_id (Optional[str]): The ID of the Airbyte workspace to load connections from. Only
+            required if multiple workspaces exist in your instance.
+        key_prefix (Optional[CoercibleToAssetKeyPrefix]): A prefix for the asset keys created.
+        create_assets_for_normalization_tables (bool): If True, assets will be created for tables
+            created by Airbyte's normalization feature. If False, only the destination tables
+            will be created.
+        connection_to_group_fn (Optional[Callable[[str], Optional[str]]]): Function which returns an asset
+            group name for a given Airbyte connection name. If None, no groups will be created. Defaults
+            to a basic sanitization function.
+    """
+
+    airbyte_instance: AirbyteResource = airbyte(build_init_resource_context())
+
+    if isinstance(key_prefix, str):
+        key_prefix = [key_prefix]
+    key_prefix = check.list_param(key_prefix or [], "key_prefix", of_type=str)
+
+    if not workspace_id:
+        workspaces = cast(
+            List[Dict[str, Any]],
+            check.not_none(airbyte_instance.make_request(endpoint="/workspaces/list", data={})).get(
+                "workspaces", []
+            ),
+        )
+
+        check.invariant(len(workspaces) <= 1, "Airbyte instance has more than one workspace")
+        check.invariant(len(workspaces) > 0, "Airbyte instance has no workspaces")
+
+        workspace_id = workspaces[0].get("workspaceId")
+
+    connections = cast(
+        List[Dict[str, Any]],
+        check.not_none(
+            airbyte_instance.make_request(
+                endpoint="/connections/list", data={"workspaceId": workspace_id}
+            )
+        ).get("connections", []),
+    )
+
+    assets = []
+    for connection_json in connections:
+        connection_id = cast(str, connection_json.get("connectionId"))
+
+        operations_json = cast(
+            Dict[str, Any],
+            check.not_none(
+                airbyte_instance.make_request(
+                    endpoint="/operations/list",
+                    data={"connectionId": connection_id},
+                )
+            ),
+        )
+        connection = AirbyteConnection.from_api_json(connection_json, operations_json)
+
+        table_mapping = connection.parse_stream_tables(create_assets_for_normalization_tables)
+
+        assets_for_connection = build_airbyte_assets(
+            connection_id=connection_id,
+            destination_tables=list(table_mapping.keys()),
+            normalization_tables=table_mapping,
+            asset_key_prefix=key_prefix,
+        )
+
+        if connection_to_group_fn:
+            assets_for_connection = list(
+                with_group(assets_for_connection, connection_to_group_fn(connection.name))
+            )
+        assets.extend(assets_for_connection)
+
+    return assets
```

Comment:
I wonder, since the user is already specifying a resource at initialization time, should it be bound as a convenience? Then all a user has to do is

`airbyte_assets = load_assets_from_airbyte_instance(...)`

instead of

`airbyte_assets = with_resources(load_assets_from_airbyte_instance(...), ...)`

If a user wants to override the resource config for run-time, they could always just use `with_resource` to set the override.

Reply:
that sounds right to me